kura/org.eclipse.kura.wire.component.provider/src/main/java/org/eclipse/kura/internal/wire/subscriber/CloudSubscriberOptions.java

```diff
@@ -66,8 +66,7 @@
     int getSubscribingQos() {
         int subscribingQos = DEFAULT_QOS;
         final Object qos = this.properties.get(CONF_QOS);
```

Comment:
You are right but this is a custom class that the OSGi runtime does not have any hold onto. Consumers of this class can pass custom properties by creating a map and pass it directly to the instance of this class which is also acceptable. I believe, it would be better to specify null checks before we process the provided map otherwise it would result in NPE.